src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,296 @@
+            final List<String> groups = getAllAnnotationGroupNames();
+            final List<String> annots = getAllAnnotationNames();
+            return new AnnotationManager(groups, annots, annotationsToExclude);
+        }
+
+        private static List<String> getAllAnnotationNames() {
+            final Set<VariantAnnotation> union = Sets.union(new HashSet<>(makeAllGenotypeAnnotations()), new HashSet<>(AnnotationManager.makeAllInfoFieldAnnotations()));
+            return union.stream().map(a -> a.getClass().getSimpleName()).collect(Collectors.toList());
+        }
+
+        /**
+         * Annotation group names are simple names of all interfaces that are subtypes of AnnotationKind.
+         */
+        public static List<String> getAllAnnotationGroupNames() {
+            final ClassFinder finder = new ClassFinder();
+            finder.find(AnnotationKind.class.getPackage().getName(), AnnotationKind.class);
+            return finder.getClasses().stream().filter(c -> c.isInterface()).map(c -> c.getSimpleName()).collect(Collectors.toList());
+        }
+
+        public List<InfoFieldAnnotation> createInfoFieldAnnotations() {
+            final List<InfoFieldAnnotation> all = makeAllInfoFieldAnnotations();
+            return filterAnnotations(all);
+        }
+
+        private static List<InfoFieldAnnotation> makeAllInfoFieldAnnotations() {
+            return Utils.makeInstances(InfoFieldAnnotation.class, AnnotationKind.class.getPackage());
+        }
+
+        public List<GenotypeAnnotation> createGenotypeAnnotations() {
+            final List<GenotypeAnnotation> all = makeAllGenotypeAnnotations();
+            return filterAnnotations(all);
+        }
+
+        private static List<GenotypeAnnotation> makeAllGenotypeAnnotations() {
+            return Utils.makeInstances(GenotypeAnnotation.class, AnnotationKind.class.getPackage());
+        }
+
+        /**
+         * Returns a list of annotations that either:
+         *  - belong to at least one of the requested annotation groups
+         *  - belong to the set of requested annotations
+         *
+         *  - and are NOT listed for exclusion
+         *
+         *  The list is sorted by simple name of the class.
+         */
+        private <T extends VariantAnnotation> List<T> filterAnnotations(final List<T> all) {
+            final SortedSet<T> annotations = new TreeSet<>(Comparator.comparing(t -> t.getClass().getSimpleName()));
+
+            for (final T t : all){
+                if (!annotationsToExclude.contains(t.getClass().getSimpleName())) {
+                    //if any group matches requested groups, it's in
+                    @SuppressWarnings("unchecked")
+                    final Set<Class<?>> superTypes = ReflectionUtils.getAllSuperTypes(t.getClass(), sup -> sup.isInterface());
+                    if (superTypes.stream().anyMatch(iface -> annotationGroupsToUse.contains(iface.getSimpleName()))) {
+                        annotations.add(t);
```

Comment:
You need to filter your super interfaces down to only those that extend `AnnotationKind` (which will hopefully get renamed to `AnnotationGroup`) before checking against `annotationGroupsToUse`